ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Tables/DeploymentVulnerabilitiesTable.tsx

```diff
@@ -63,19 +63,23 @@ export type DeploymentWithVulnerabilities = {
     }[];
 };
 
+type DeploymentVulnerabilityImageMapping = {
+    imageMetadataContext: ImageMetadataContext;
+    componentVulnerabilities: DeploymentComponentVulnerability[];
+};
+
 function formatVulnerabilityData(deployment: DeploymentWithVulnerabilities): {
     cve: string;
     severity: VulnerabilitySeverity;
     isFixable: boolean;
     discoveredAtImage: Date | null;
     summary: string;
     affectedComponentsText: string;
-    images: {
-        imageMetadataContext: ImageMetadataContext;
-        componentVulnerabilities: DeploymentComponentVulnerability[];
-    }[];
+    images: DeploymentVulnerabilityImageMapping[];
 }[] {
-    const imageMap: Record<string, ImageMetadataContext> = {};
+    // Create a map of image ID to image metadata for easy lookup
```

Comment:
This function receives data from a GraphQL query will the following basic structure:
```graphql
query getCvesForDeployment(
	$id: ID!
	$query: String!
	$pagination: Pagination!
) {
	deployment(id: $id) {
		id
		images(query: $query) {
			id
			metadata {
				v1 {
					layers { instruction value }
				}
			}
		}
		imageVulnerabilities(query: $query, pagination: $pagination) {
			cve
			images(query: $query) {
				imageId: id
				imageComponents(query: $query) {
					name
					layerIndex
					imageVulnerabilities(query: $query) {
						vulnerabilityId: id
					}
				}
			}
		}
	}
}
```
At the top level, we query for `deployments->images` to get a list of all images associated with the deployment, and their layer metadata. Deeper in the query we query for all images again, but specific to `imageVulnerabilities` that are part of the deployment (`deployment->images->imageVulnerabilities->`images`). This is because we need the full flattened list of `vuln + image + component` for display in the table.

When we display the data client side we need to match the "inner" images with the "outer" images and metadata to get the correct line from the Dockerfile. In order to do so we create a mapping of IDs->Images with `Record<string, ImageMetadataContext>`.

The problem with this, is that without the tsconfig `noUncheckedIndexAccess`, indexing into the above object with any string always returns a type of `ImageMetadataContext`. This is obviously not correct since we cannot have a valid value for every possible `string`.

The error comes from when we have an "inner" image with an id that doesn't exist in the "outer" images. This results in an `undefined` `ImageMetadataContext` which is passed further down in the code, and eventually errors out when we try to access the `metadata` property. This could be from a mismatch in data returned from the server, some issue with the Apollo cache, or something else. (We still have not reproduced.)

The "fix" here is to just do an additional check to ensure that this object is never null.